app/services/data_exports/csv/invoice_fees.rb

```diff
@@ -96,7 +46,55 @@ def self.headers
 
       private
 
-      attr_reader :invoice_serializer_klass, :fee_serializer_klass, :subscription_serializer_klass
+      attr_reader :data_export_part, :invoice_serializer_klass, :fee_serializer_klass
+
+      def serialized_item(fee)
+        serialized_fee = fee_serializer_klass.new(fee).serialize
+        serialized_invoice = invoice_serializer_klass.new(fee.invoice).serialize
+
+        serialized_subscription = {
+          external_id: fee.subscription&.external_id,
+          plan_code: fee.subscription&.plan&.code
+        }
+
+        [
+          serialized_invoice[:lago_id],
+          serialized_invoice[:number],
+          serialized_invoice[:issuing_date],
+          serialized_fee[:lago_id],
+          serialized_fee.dig(:item, :type),
+          serialized_fee.dig(:item, :code),
+          serialized_fee.dig(:item, :name),
+          serialized_fee.dig(:item, :description),
+          serialized_fee.dig(:item, :invoice_display_name),
+          serialized_fee.dig(:item, :filter_invoice_display_name),
+          serialized_fee.dig(:item, :grouped_by),
+          serialized_subscription[:external_id],
+          serialized_subscription[:plan_code],
+          serialized_fee[:from_date],
+          serialized_fee[:to_date],
+          serialized_fee[:total_amount_currency],
+          serialized_fee[:units],
+          serialized_fee[:precise_unit_amount],
+          serialized_fee[:taxes_amount_cents],
+          serialized_fee[:total_amount_cents]
+        ]
+      end
+
+      def collection
+        Fee
```

Comment:
This could loads a lot of records, this is why it's done per invoice bases, we have memory issues in the past.

Reply:
I've reverted these changes to keep original approach. Not ideal but from performance standpoint should be the same.